packages/taraxa-ui/src/components/BlockTable/BlockTable.tsx

```diff
@@ -0,0 +1,250 @@
+import React from 'react';
+import {
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  Box,
+  TablePagination,
+  TableContainer,
+  ThemeProvider,
+  CssBaseline,
+} from '@mui/material';
+import Label from '../Label';
+import Button from '../Button';
+import {
+  TransactionIcon,
+  GreenRightArrow,
+  GreenCircledCheck,
+  RedCircledCancel,
+  TransactionBlock,
+} from '../Icons';
+import theme from '../theme';
+
+export enum TransactionStatus {
+  SUCCESS = '0x0',
+  FAILURE = '0x1',
+  IN_PROGRESS = '0x',
+}
+
+export interface TransactionData {
+  txHash: string;
+  status: TransactionStatus;
+  timestamp: string;
+  pbftBlock: string;
+  dagLevel: string;
+  dagHash: string;
+  value: string;
+  from: string;
+  gasLimit: string;
+  gas: string;
+  gasPrice: string;
+  to: string;
+  nonce: number;
+  transactionLink: JSX.Element;
+  addressLinkFrom: JSX.Element;
+  addressLinkTo: JSX.Element;
+}
+
+export const statusToLabel = (state: TransactionStatus) => {
+  if (state === TransactionStatus.SUCCESS) {
+    return (
+      <Label
+        variant='success'
+        label='Success'
+        gap
+        icon={<GreenCircledCheck />}
+      />
+    );
+  }
+  if (state === TransactionStatus.FAILURE) {
+    return (
+      <Label variant='error' label='Failure' gap icon={<RedCircledCancel />} />
+    );
+  }
+  return (
+    <Label variant='error' label='Failure' gap icon={<RedCircledCancel />} />
```

Comment:
This creates a problem in the UI. While the data is loading, this shows as Error and then changes to it's correct status. We should have a Loading label, with a grey color showing a small spinner. This needs to be fixed here and also in the explorer where this is used.